src/lsqecc/pauli_rotations/circuit.py

```diff
@@ -209,35 +256,12 @@ def commute_pi_over_four_rotation(self, index: int) -> None:
         }:
             raise Exception("First operand must be +-pi/4 Pauli rotation")
 
-        # Need to calculate iPP' when PP' = -P'P (anti-commute)
         if not PauliOpCircuit.are_commuting(self.ops[index], self.ops[next_block]):
-            product_of_coefficients = complex(1)
-
-            for i in range(self.qubit_num):
-                new_op = PauliOperator.multiply_operators(
-                    self.ops[index].get_op(i), self.ops[next_block].get_op(i)
-                )
-
-                self.ops[next_block].change_single_op(i, new_op[1])
-                product_of_coefficients *= new_op[0]
-
-            # Flip the phase if product of coefficients is negative
-            # Product of coefficients will always be either i or -i (see issues #28 for proof)
-            product_of_coefficients *= 1j
-            if isinstance(self.ops[next_block], Measurement):
-                if product_of_coefficients.real < 0:
-                    measurement = cast(Measurement, self.ops[next_block])
-                    measurement.isNegative = not measurement.isNegative
-
-            else:
-                cast(PauliRotation, self.ops[next_block]).rotation_amount *= (
-                    -1 if product_of_coefficients.real < 0 else 1
-                )
-
-        temp = self.ops[index]
-        self.ops[index] = self.ops[next_block]
-        self.ops[next_block] = temp
-        # print(self.render_ascii())
+            self._swap_adjacent_anticommuting_blocks(index)
+        elif PauliOpCircuit.are_commuting(self.ops[index], self.ops[next_block]):
+            self._swap_adjacent_commuting_blocks(index)
+        else:
+            raise Exception("The two blocks must commute or anti-commute!")
```

Comment:
```suggestion
        else:
            self._swap_adjacent_commuting_blocks(index)
```
`are_commuting` will only return True or False so this is unnecessary. We will do the error handling in there.

Reply:
Made the suggested changes.